From 203a59d470f624301edd3dce17b961f9f99511db Mon Sep 17 00:00:00 2001
From: "djm@openbsd.org" <djm@openbsd.org>
Date: Sun, 28 Jun 2026 23:47:16 +0000
Subject: [PATCH 02/12] upstream: resist that return ".." via remote glob
 during

remote/remote copies, similar to fixes for bz3871 for remote/local copies.
From Swival scanner

OpenBSD-Commit-ID: c0c20a1b746db55c08e53658bf21ea9405b300a5
(cherry picked from commit 36480181fa22f98e180b4f9e10203480c0346c78)
[CVE-2026-59996]
---
 scp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scp.c b/scp.c
index 1faa9a555744..bd01ebffd936 100644
--- a/scp.c
+++ b/scp.c
@@ -2039,6 +2039,10 @@ throughlocal_sftp(struct sftp_conn *from, struct sftp_conn *to,
 			goto out;
 		}
 
+		/* Special handling for source of '..' */
+		if (strcmp(filename, "..") == 0)
+			filename = "."; /* Download to dest, not dest/.. */
+
 		if (targetisdir)
 			abs_dst = sftp_path_append(target, filename);
 		else
-- 
2.55.0.dirty

